sdk/servicebus/service-bus/test/stress/stress_lockRenewal.ts

```diff
@@ -0,0 +1,107 @@
+/*
+Test Scenario summary:
+- Creates a single sender and a single receiver on a queue.
+- Receives a message and holds onto it for the duration of test. AutoLockRenewal is enabled and set to the test duration.
+
+The test assumes no other process is working with the queues defined in here,
+but the queues must be empty and use default configurations before running the test.
+
+For running this test, connection string of the Service Bus namespace and queue name
+must be supplied.
+*/
+
+import {
+  ServiceBusClient,
+  SendableMessageInfo,
+  OnMessage,
+  OnError,
+  delay,
+  ReceiveMode,
+  ServiceBusMessage
+} from "../../src";
+
+const connectionString = "";
+const queueName = "";
+
+const testDurationInMilliseconds = 60000 * 5 * 12 * 24; // 24 hours
+
+let snapshotIntervalID: any;
+
+let receivedMessage: ServiceBusMessage;
+
+async function main(): Promise<void> {
+  snapshotIntervalID = setInterval(snapshot, 60000); // Every 60 seconds
+  await sendMessage();
+  await receiveMessage();
+}
+
+async function sendMessage(): Promise<void> {
+  const ns = ServiceBusClient.createFromConnectionString(connectionString);
+  const client = ns.createQueueClient(queueName);
+  try {
+    const sender = client.createSender();
+
+    const message: SendableMessageInfo = {
+      messageId: "test",
+      body: "test",
+      label: `test`
+    };
+
+    await sender.send(message);
+  } finally {
+    await client.close();
+    await ns.close();
+  }
+}
+
+async function receiveMessage(): Promise<void> {
+  const ns = ServiceBusClient.createFromConnectionString(connectionString);
+  const client = ns.createQueueClient(queueName);
+
+  try {
+    const receiver = client.createReceiver(ReceiveMode.peekLock);
+    const onMessageHandler: OnMessage = async (brokeredMessage) => {
+      receivedMessage = brokeredMessage;
+      if (receivedMessage.messageId !== "test") {
+        throw new Error("Message is corrupt or unexpected");
+      }
+      console.log("Received message: ", receivedMessage.messageId);
+
+      await delay(testDurationInMilliseconds);
+      await brokeredMessage.complete();
+      console.log("Completed message: ", receivedMessage.messageId);
+    };
+
+    const onErrorHandler: OnError = (err) => {
+      throw err;
```

Comment:
Did we test if this error surfaces up well and gets caught in the `main().catch()` ?

Reply:
Since all we are doing in `main().catch(0` is to `console.log` the error, why not console.log here instead of throwing?

Reply:
Yes, the error doesn't surface up. We had discussed this scenario yesterday (in a slightly different context) - that the onError handler is supposed to be the terminal executing code.
The process crashes with an error not handled exception.

Idea, is that the test should run without any error, so it would be necessary to not swallow and know about it.

Reply:
Got it, thanks

Reply:
Can you add a comment above the throw regarding this?